src/pyk/kcfg/kcfg.py

```diff
@@ -654,6 +620,18 @@ def replace_node(self, node_id: str, new_cterm: CTerm) -> str:
 
         return new_node.id
 
+    def _assert_no_successors(self, source_id: str) -> None:
```

Comment:
`_assert` suggest an `AssertionError` (and thus a bug), consider `_check` instead:

```suggestion
    def _check_no_successors(self, source_id: str) -> None:
```

Reply:
Done.